### PR TITLE
Miscellaneous Fixes for Suggested Sites

### PR DESCRIPTION
**Pull request by evancrow**
### How was this tested?
- [ ] I tested this manually
- [x] I added automated tests
- [x] Existing automated tests are enough to cover my changes

## Issues addressed
Fixed: #3135
Fixed: #3070

### CODE REVIEW

File: Client/Frontend/ZeroQuery/ZeroQueryModel.swift [hunk 1]
```diff
@@ -240,15 +240,16 @@ class ZeroQueryModel: ObservableObject {
...
-        // if the default top sites contains the siteurl. also wipe it from default suggested sites.
+        // If the default top sites contains the site URL, also wipe it from default suggested sites.
         if TopSitesHandler.defaultTopSites().filter({ $0.url == url }).isEmpty == false {
             Defaults[.deletedSuggestedSites].append(url.absoluteString)
         }
+        
         profile.history.removeHostFromTopSites(host).uponQueue(.main) { result in
             guard result.isSuccess else { return }
             self.profile.panelDataObservers.activityStream.refreshIfNeeded(forceTopSites: true)
-            self.updateSuggestedSites()
```
Comment (chung-lai) [inline Client/Frontend/ZeroQuery/ZeroQueryModel.swift:251]: Was this fixing suggested site not immediately removed from the screen? We don't need this anymore?

**Review by chung-lai — COMMENTED** [review]
Just making sure I understand the change, we don't use autolayout for `FadingHorizontalScrollView` and let system to do autosizing?

File: Client/Frontend/ZeroQuery/ZeroQueryModel.swift (hunk 1 above)
Comment (evancrow) [reply Client/Frontend/ZeroQuery/ZeroQueryModel.swift:251]: So what happened is that the actual updating of the `SuggestedSites` takes a while so it quickly returned the old values. So if we just delete it from the array instantly, and then update the array in the background for when the app actually updates it works.

**Comment by evancrow** [comment]
> Just making sure I understand the change, we don't use autolayout for `FadingHorizontalScrollView` and let system to do autosizing?

We're still doing what we're always doing, using the UIKit layout constraints. But instead of attaching to the `superview` which is sometimes nil, we attach to the view it's actually embedded in.

**Review by chung-lai — APPROVED** [review]
LGTM, thanks!

**Pull request merged**